services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -593,12 +601,42 @@ protected void initDataColumnSidecarManager() {
       DataColumnSidecarValidator dataColumnSidecarValidator = DataColumnSidecarValidator.create();
       DataColumnSidecarGossipValidator gossipValidator =
           DataColumnSidecarGossipValidator.create(dataColumnSidecarValidator);
-      dataColumnSidecarManager = DataColumnSidecarManager.create(gossipValidator);
+      dataColumnSidecarManager = new DataColumnSidecarManagerImpl(gossipValidator);
+      eventChannels.subscribe(
+          DataColumnSidecarGossipChannel.class,
+          dataColumnSidecarManager::onDataColumnSidecarPublish);
     } else {
       dataColumnSidecarManager = DataColumnSidecarManager.NOOP;
     }
   }
 
+  protected void initDasCustody() {
+    DataColumnSidecarDBImpl sidecarDB =
+        new DataColumnSidecarDBImpl(
+            combinedChainDataClient, eventChannels.getPublisher(SidecarUpdateChannel.class));
+    DataColumnSidecarCustodyImpl.BlockChainAccessor blockChainAccessor =
```

Comment:
maybe give some speaking name, say `blockRootResolver` or whatever. BlockChainAccessor is too broad